### PR TITLE
added lesson 12: debugging

### PR DESCRIPTION
**Pull request by 3tom**
Dear @tomekDiakowTieto ,

Please take a look.

Thanks,
Tomasz

### CODE REVIEW

File: students/medrek_tomasz/lesson_12_debugging/regex_search.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    for file in os.listdir(dir_path):
+        if file.endswith(".txt"):
+            count_files += 1
+
+            try:
+                fh = open(os.path.join(dir_path, file), 'r')
+                string_from_file = fh.readlines()
+                fh.close()
+            except (UnicodeDecodeError, PermissionError) as e:
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_12_debugging/regex_search.py]: https://github.com/jedzej/tietopythontraining-basic/issues/606

File: students/medrek_tomasz/lesson_12_debugging/validated_user_base.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+
+import sys
+import os
+import shutil
+import argparse
+import logging
+import csv
+sys.path.append('..')
+from lesson_08_regular_expressions import email_validator,\
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_12_debugging/validated_user_base.py]: How can sys.path.append('..') be replaced then?

File: students/medrek_tomasz/lesson_12_debugging/validated_user_base.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        logger.setLevel(logging.WARNING)
+    elif dbg_lvl == 0:
+        logging.disable(logging.CRITICAL)
+
+
+def combined_validator(dict_entry):
+    for x, y in dict_entry.items():
+        globals()[x] = y
+
+    if not email_validator.email_validator(email):
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_12_debugging/validated_user_base.py]: yes, but I like it better than using dictionary every time:
for x, y in dict_entry.items():
    globals()[x] = y

File: students/medrek_tomasz/lesson_12_debugging/validated_user_base.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        raise ValidationError(email)
+    if not strong_password_detection.strong_password_detection(password):
+        raise ValidationError(password)
+    if not phone_number_validator.phone_number_validator(phone_number):
+        raise ValidationError(phone_number)
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_12_debugging/validated_user_base.py]: yes, but I like it better than using dictionary every time:
for x, y in dict_entry.items():
    globals()[x] = y

File: students/medrek_tomasz/lesson_12_debugging/validated_user_base.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    if not postal_code_validator_poland.postal_code_validator_poland(
+            postal_code):
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_12_debugging/validated_user_base.py]: yes, but I like it better than using dictionary every time:
for x, y in dict_entry.items():
    globals()[x] = y

File: students/medrek_tomasz/lesson_12_debugging/validated_user_base.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        raise ValidationError(postal_code)
```
[the remark this answers is not in the document]
Comment (3tom) [reply students/medrek_tomasz/lesson_12_debugging/validated_user_base.py]: yes, but I like it better than using dictionary every time:
for x, y in dict_entry.items():
    globals()[x] = y

File: students/medrek_tomasz/lesson_12_debugging/validated_user_base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomekDiakowTieto) [reply students/medrek_tomasz/lesson_12_debugging/validated_user_base.py]: You can use:
https://docs.python.org/2/tutorial/modules.html#intra-package-references
look at the example:
from . import echo
from .. import formats
from ..filters import equalizer

**Pull request merged**